pynamodb/attributes.py

```diff
@@ -137,7 +135,8 @@ def _get_attributes(cls):
 
         :rtype: dict[str, Attribute]
         """
-        if cls._attributes is None:
+        if '_attributes' not in cls.__dict__:
```

Comment:
`hasattr` is equivalent to calling `getattr(cls, '_attributes'` which first searches the class and if now found continues to try to resolve the name by searching the class tree. The result would be that `hasattr` would return True for all subclasses of `MapAttribute` as soon as any instance had been created (and `getattr` would return the empty dictionary that is `MapAttribute._attributes`). What we actually want here is for each and every subclass of `AttributeContainer` to have their own `_attributes` dictionary based on the class definition.